Source/WebKit/Platform/IPC/ArgumentCoders.h

```diff
@@ -499,6 +499,61 @@ template<typename KeyType, typename ValueType> struct ArgumentCoder<WTF::KeyValu
     }
 };
 
+template<typename T, size_t size> struct ArgumentCoder<std::array<T, size>> {
+    template<typename Encoder>
+    static void encode(Encoder& encoder, const std::array<T, size>& array)
+    {
+        for (auto& item : array)
+            encoder << item;
+    }
+
+    template<typename Decoder>
+    static std::optional<std::array<T, size>> decode(Decoder& decoder)
+    {
+        std::array<std::optional<T>, size> items;
+
+        for (auto& item : items) {
```

Comment:
Hey, that's nifty. The caller indicates how many items they expect, so there isn't a need to send the size across the wire.

Reply:
Yes, it's encoded in the type itself.